be/src/formats/csv/converter.cpp

```diff
@@ -56,8 +57,24 @@ static std::unique_ptr<Converter> get_converter(const TypeDescriptor& t) {
         return std::make_unique<DateConverter>();
     case TYPE_DATETIME:
         return std::make_unique<DatetimeConverter>();
-    case TYPE_ARRAY:
-        return std::make_unique<ArrayConverter>(get_converter(t.children[0], true));
+    case TYPE_ARRAY: {
```

Comment:
Codepaths for hive and broker load are almost identical. It could be more compact
```cpp
if (c == nullptr) {
    return is_hive ? std::make_unique<DefaultValueConverter>() : nullptr;
}
```

Reply:
improved